### PR TITLE
Allow not-escaping the value in set_value()

### PR DESCRIPTION
**Pull request by avenirer**
Wouldn't it be a good idea to allow for not escaped html in set_value()?

### CODE REVIEW

File: system/helpers/form_helper.php [hunk 1]
```diff
@@ -676,17 +676,18 @@ function form_prep($str)
 	 *
 	 * @param	string	$field		Field name
 	 * @param	string	$default	Default value
+	 * @param	bool	$escaped	Escaped value
 	 * @return	string
 	 */
-	function set_value($field, $default = '')
+	function set_value($field, $default = '',$escaped = TRUE)
```
Comment (narfbg) [inline system/helpers/form_helper.php]: Needs a space after the comma, and should be named `$escape` (no 'd').

File: system/helpers/form_helper.php [hunk 2]
```diff
@@ -676,17 +676,18 @@ function form_prep($str)
...
 	{
 		$CI =& get_instance();
 
 		$value = (isset($CI->form_validation) && is_object($CI->form_validation) && $CI->form_validation->has_rule($field))
 			? $CI->form_validation->set_value($field, $default)
 			: $CI->input->post($field, FALSE);
-
-		return html_escape($value === NULL ? $default : $value);
+		$value = ($value === NULL) ? $default : $value;
```
Comment (narfbg) [inline system/helpers/form_helper.php]: Might as well change that to:

    isset($value) OR $value = $default;

Also, please don't remove the empty line above it.

File: system/helpers/form_helper.php [hunk 3]
```diff
@@ -676,17 +676,18 @@ function form_prep($str)
...
+		return $escaped ? html_escape($value) : $value;
```
Comment (narfbg) [inline system/helpers/form_helper.php]: When using a ternary with a variable, please wrap it in parenthesis - makes it more readable.

**Comment by narfbg** [comment]
Also needs a changelog entry and a documentation update.

File: system/helpers/form_helper.php [hunk 4]
```diff
@@ -676,17 +676,19 @@ function form_prep($str)
 	 *
 	 * @param	string	$field		Field name
 	 * @param	string	$default	Default value
+	 * @param	bool	$html_escape	HTML escaped value
```
Comment (narfbg) [inline system/helpers/form_helper.php]: "Whether to escape HTML special characters" or something like that ... "HTML escaped value" means a completely different thing.

**Comment by avenirer** [comment]
done...

**Comment by narfbg** [comment]
Great, just one change on the docblock remains ... :)

**Comment by avenirer** [comment]
@narfbg I really don't envy you. You must have nerves of steel to be so patient... :+1:

**Comment by narfbg** [comment]
Thanks. :)

**Pull request merged**